subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/junit/JUnitCategoriesCoverageIntegrationSpec.groovy

```diff
@@ -75,6 +77,100 @@ class JUnitCategoriesCoverageIntegrationSpec extends JUnitMultiVersionIntegratio
+        result.assertTestClassesExecuted('org.gradle.CatATests', 'org.gradle.CatBTests', 'org.gradle.CatADTests', 'org.gradle.MixedTests')
+        result.testClass("org.gradle.CatATests").assertTestCount(4, 0, 0)
+        result.testClass("org.gradle.CatATests").assertTestsExecuted('catAOk1[0]', 'catAOk2[0]', 'catAOk3[0]', 'catAOk4[0]')
+        result.testClass("org.gradle.CatBTests").assertTestCount(4, 0, 0)
+        result.testClass("org.gradle.CatBTests").assertTestsExecuted('catBOk1[0]', 'catBOk2[0]', 'catBOk3[0]', 'catBOk4[0]')
+        result.testClass("org.gradle.CatADTests").assertTestCount(2, 0, 0)
+        result.testClass("org.gradle.CatADTests").assertTestsExecuted('catAOk1[0]', 'catAOk2[0]')
+        result.testClass("org.gradle.MixedTests").assertTestCount(3, 0, 0)
+        result.testClass("org.gradle.MixedTests").assertTestsExecuted('catAOk1[0]', 'catBOk2[0]')
+        result.testClass("org.gradle.MixedTests").assertTestsSkipped('someIgnoredTest[0]')
+    }
+
+    @Issue('https://github.com/gradle/gradle/issues/1153')
+    def canSpecifyExcludesOnlyWithParameterized() {
+        given:
+        resources.maybeCopy('JUnitCategoriesCoverageIntegrationSpec/canSpecifyExcludesOnly')
+        configureJUnit()
+        rewriteTestWithParameterized()
+
+        when:
+        run('test')
+
+        then:
+        DefaultTestExecutionResult result = new DefaultTestExecutionResult(testDirectory)
+        result.assertTestClassesExecuted('org.gradle.NoCatTests', 'org.gradle.SomeTests', 'org.gradle.SomeOtherCatTests')
+        result.testClass("org.gradle.SomeOtherCatTests").assertTestCount(2, 0, 0)
+        result.testClass("org.gradle.SomeOtherCatTests").assertTestsExecuted('someOtherOk1[0]', 'someOtherOk2[0]')
+        result.testClass("org.gradle.NoCatTests").assertTestCount(2, 0, 0)
+        result.testClass("org.gradle.NoCatTests").assertTestsExecuted('noCatOk1[0]', 'noCatOk2[0]')
+        result.testClass("org.gradle.SomeTests").assertTestCount(3, 0, 0)
+        result.testClass("org.gradle.SomeTests").assertTestsExecuted('noCatOk3[0]', 'noCatOk4[0]', 'someOtherCatOk2[0]')
+    }
+
+
+    /*
+    This method converts a non-parameterized test class to the parameterized version without changing its behaviour
+    For example,
+
+    package org.gradle;
+    public class CatTests {
+        @Test public void ok() {}
+    }
+
+    will be converted to
+
+    package org.gradle;
+    import org.junit.runners.Parameterized;
+    import org.junit.runner.RunWith;
+    import java.util.Arrays;
+
+    @RunWith(Parameterized.class)
+    public class CatTests {
+        @Parameterized.Parameter
+        public String name;
+
+        @Parameterized.Parameters
+        public static Iterable<Object[]> getParameters() { return Arrays.asList(new Object[] { "1" }); }
+
+        @Test public void ok() {}
+    }
+    */
+
+    def rewriteTestWithParameterized() {
```

Comment:
❌ Instead of doing some magic translation here, we should just write a dedicated test for this.